src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/authorization/UserAccessScopeMapper.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.DynamicFrameworkContract
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ServiceProvider
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.DynamicFrameworkContractRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ServiceProviderRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.HMPPSAuthService
+import java.lang.RuntimeException
+
+data class UserAccessScope(
+  val serviceProvider: ServiceProvider,
+  val contracts: List<DynamicFrameworkContract>,
+)
+
+@Component
+class UserAccessScopeMapper(
+  private val hmppsAuthService: HMPPSAuthService,
+  private val serviceProviderRepository: ServiceProviderRepository,
+  private val dynamicFrameworkContractRepository: DynamicFrameworkContractRepository,
+) {
+  private val serviceProviderGroupPrefix = "INT_SP_"
+  private val contractGroupPrefix = "INT_CG_"
+
+  // get the user access scope for a given service provider user
+  fun forServiceProviderUser(user: AuthUser): UserAccessScope {
```

Comment:
What do you think about `validatedScopeFor(user)`?

Reply:
would you keep the name of the class as it?

`userAccessScopeMapper.validatedScopeFor` ?

i think it's important here to distinguish between sp and pp users ( since pp users have no concept of an access scope).

Reply:
👍 `providerScopeMapper` maybe?

Should we have a different `AuthUser` class for SPs and PPs to make this explicit?

Reply:
after our discussion we agreed to rename the whole component to be SP specific. i've made this change now.